Lib/fontTools/ufoLib/filenames.py

```diff
@@ -15,7 +15,7 @@ class NameTranslationError(Exception):
 	pass
 
 
-def userNameToFileName(userName: str, existing=[], prefix="", suffix=""):
+def userNameToFileName(userName: str, existing=set(), prefix="", suffix=""):
```

Comment:
avoid the mutable default value here by setting `existing=None` and creting the empty set within the function body `if existing is None`

Reply:
Or default to an empty tuple. But it's a redundant change as `existing` will never be modified.

Reply:
> Hm, this function might be called from outside, so I'll have to guard against None inside.

it was not checking for None-ness before when it was defaulting to an empty (yet mutable) list. I don't think you need to guard for None.